styx-scheduler-service/src/main/java/com/spotify/styx/Scheduler.java

```diff
@@ -221,13 +223,15 @@ private void dequeueInstances(
       Map<WorkflowId, Workflow> workflows,
       List<InstanceState> eligibleInstances) {
 
+    final ConcurrentMap<String, Boolean> resourceExhaustedCache = new ConcurrentHashMap<>();
```

Comment:
Where do we invalidate the cache?

Reply:
A new instance is created for for every tick.

Reply:
Ah ok i read it as instance variable.